include/ginkgo/core/preconditioner/ic.hpp

```diff
@@ -142,14 +141,16 @@ class Ic : public EnableLinOp<Ic<LSolverType, IndexType>>, public Transposable {
 
         GKO_DEPRECATED("use with_l_solver instead")
         parameters_type& with_l_solver_factory(
-            deferred_factory_parameter<const typename l_solver_type::Factory>
+            deferred_factory_parameter<
+                const gko::detail::factory_type<l_solver_type>>
                 solver)
         {
             return with_l_solver(std::move(solver));
         }
 
         parameters_type& with_l_solver(
-            deferred_factory_parameter<const typename l_solver_type::Factory>
+            deferred_factory_parameter<
+                const gko::detail::factory_type<l_solver_type>>
```

Comment:
I think this function needs some documentation to clarify what the allowed inputs are.